File: Anul II/Medii de proiectare si programare/Laborator/lab09/services/src/main/java/com/ubb/service/IService.java
package com.ubb.service;

import com.ubb.model.Identifiable;

import java.util.Collection;
import java.util.Optional;

public interface IService<ID, T extends Identifiable<ID>> {
    Optional<T> findById(final ID id);

    Collection<T> findAll();

    void save(final T newEntity);

    void delete(final ID id);

    void update(final T updatedEntity);
}

File: Anul II/Medii de proiectare si programare/Laborator/lab08/C#/Ubb.BikeContest.Service/src/IService.cs
using System.Collections;
using Ubb.BikeContest.Model;

namespace Ubb.BikeContest.Service;

public interface IService<TId, TEntity> where TEntity : Identifiable<TId>
{
    TEntity FindById(TId id);

    IEnumerable<TEntity> FindAll();

    void Save(TEntity newEntity);

    void Delete(TId id);

    void Update(TEntity updatedEntity);
}